### PR TITLE
18 auth screens

### LINKED ISSUE #18: Auth Screens
- Implement the UI for Splash Screen and GetStarted Screen
- Login button should navigate the user to GetStarted Screen
- Get Started button should navigate to Home Screen

### PR DESCRIPTION
**Pull request by Amore174**
Implemented UI for the splash screen and get started screen and also navigation for both.

### CODE REVIEW

File: lib/screens/get_started/get_started.dart [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import 'package:go_router/go_router.dart';
+import 'package:icragee_mobile/screens/splash.dart';
+class GetStarted extends StatelessWidget {
+  const GetStarted({super.key});
+
+  //BuildContext get context => null;
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
```
Comment (hardikroongta8) [inline lib/screens/get_started/get_started.dart]: Dont use Material App
its already at the root of the app

File: lib/screens/get_started/get_started.dart [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        debugShowCheckedModeBanner: false,
+      home: buildScaffold(context),
+
+    );
+  }
+
+  Scaffold buildScaffold(BuildContext context) {
```
Comment (hardikroongta8) [inline lib/screens/get_started/get_started.dart]: dont use functional widgets
create a new stateless/stateful widget if required

File: lib/screens/splash.dart [hunk 3]
```diff
@@ -1,11 +1,62 @@
...
+import 'package:icragee_mobile/screens/get_started/get_started.dart';
 import 'package:icragee_mobile/screens/home_screen.dart';
 
 class SplashScreen extends StatelessWidget {
   const SplashScreen({super.key});
 
   @override
   Widget build(BuildContext context) {
-    return const HomeScreen();
+    return  MaterialApp(
```
Comment (hardikroongta8) [inline lib/screens/splash.dart]: remove material app

File: lib/screens/splash.dart [hunk 4]
```diff
@@ -1,11 +1,62 @@
...
+
+            const SizedBox(height: 24),
+            buildElevatedButton(context,"Log in as Guest"),],
+        )
+
+      ),
+    );
+  }
+
+  ElevatedButton buildElevatedButton(BuildContext context, String buttonName) {
```
Comment (hardikroongta8) [inline lib/screens/splash.dart]: create a stateless widget for this

**Review by hardikroongta8 — CHANGES_REQUESTED** [review]

File: lib/screens/splash.dart (hunk 4 above)
Comment (Amore174) [reply lib/screens/splash.dart]: done

File: lib/screens/splash.dart (hunk 3 above)
Comment (Amore174) [reply lib/screens/splash.dart]: done

File: lib/screens/get_started/get_started.dart (hunk 2 above)
Comment (Amore174) [reply lib/screens/get_started/get_started.dart]: done

File: lib/screens/get_started/get_started.dart (hunk 1 above)
Comment (Amore174) [reply lib/screens/get_started/get_started.dart]: done

**Review by hardikroongta8 — APPROVED** [review]

**Pull request merged**